### PR TITLE
Finish

### PR DESCRIPTION
**Pull request by ZVIproject**

### CODE REVIEW

File: src/main/java/com/zviproject/component/entity/Message.java [hunk 1]
```diff
@@ -10,59 +10,78 @@
...
 	@GeneratedValue(strategy = GenerationType.IDENTITY)
-	private Integer id;
+	private Integer id_message;
+	private int id_sender;
+	private int id_receiver;
+	private Date send_time;
+	private String body;
+	private Date date_time;
+	private Date time_stamp;
+
```
Comment (OlegNovatskiy) [inline src/main/java/com/zviproject/component/entity/Message.java:33]: Name of properties must by UperCase

**Review by af150416 — APPROVED** [review]

File: src/main/java/com/zviproject/component/entity/MessageToDisplay.java [hunk 2]
```diff
@@ -22,17 +23,23 @@
...
 	private int id;
 
-	private String text;
+	@Column(name = "body")
+	private String textMessage;
+
 	private String senderTok;
+
+	@Column(name = "send_time")
 	private Date time;
```
Comment (OlegNovatskiy) [inline src/main/java/com/zviproject/component/entity/MessageToDisplay.java]: Bad name - time. What is a time? Maybe there must by timesendMessage?

File: src/main/java/com/zviproject/component/entity/User.java [hunk 3]
```diff
@@ -9,24 +10,32 @@
...
 
 	@Id
 	@GeneratedValue(strategy = GenerationType.IDENTITY)
-	private Integer id;
-
+	private Integer id_user;
+	private String login;
 	private String access_token;
+	private Date date_time;
+	private Date time_stamp;
```
Comment (OlegNovatskiy) [inline src/main/java/com/zviproject/component/entity/User.java:24]: use upercase

File: src/main/java/com/zviproject/component/service/ChatDao.java [hunk 4]
```diff
@@ -19,130 +20,77 @@
...
+	 * Get short information about message
 	 */
-	private static Logger log = Logger.getLogger(ChatDao.class.getName());
+	private static final String SQL_INFORMATION_ABOUT_USERS = "SELECT m.body, m.send_time, m.id_message, s.access_token AS senderTok"
+			+ " FROM messages AS m"
+			+ " LEFT JOIN users AS r ON r.id_user=m.id_receiver LEFT JOIN users AS s ON s.id_user = m.id_sender"
+			+ " WHERE (m.id_sender = :senderId AND m.id_receiver=:receiverId) OR (m.id_sender=:receiverId AND m.id_receiver=:senderId)";
 
-	
+	private static final String SQL_COUNT_USERS = "SELECT COUNT(id_user) FROM users WHERE users.id_user LIKE (:senderId) OR  users.id_user LIKE(:receiverId)";
```
Comment (OlegNovatskiy) [inline src/main/java/com/zviproject/component/service/ChatDao.java]: "LIKE" is bad, you must use here '='

File: src/main/java/com/zviproject/component/service/ChatDao.java [hunk 5]
```diff
@@ -189,45 +133,101 @@ public int sendMessage(int sender, int receiver, String text) {
...
-	public int registerUser(User user) {
+	public ReturnedId registerUser(User user) {
 		Session session = hibernateUtil.getSessionFactory().openSession();
-		boolean accessReg = registerUser(user.getName());
-		if (accessReg) {
-		session.save(user);
+		boolean accessToRegister = checkRegisterUser(user.getLogin());
 
-		int id = user.getId();
+		if (accessToRegister) {
```
Comment (OlegNovatskiy) [inline src/main/java/com/zviproject/component/service/ChatDao.java]: You can change it 'if' statement

Comment (OlegNovatskiy) [reply src/main/java/com/zviproject/component/service/ChatDao.java]: if (!accessToRegister) {
returnedId.setStatus("error");
return returnedId;
}
session.save(user);
returnedId.setId(user.getId_user());
......
return returnedId;

File: src/main/java/com/zviproject/component/entity/Message.java (hunk 1 above)
Comment (af150416) [reply src/main/java/com/zviproject/component/entity/Message.java:33]: Why?

Comment (af150416) [reply src/main/java/com/zviproject/component/entity/Message.java:33]: only constant in uperCase

Comment (ZVIproject) [reply src/main/java/com/zviproject/component/entity/Message.java:33]: Done

File: src/main/java/com/zviproject/component/entity/MessageToDisplay.java (hunk 2 above)
Comment (ZVIproject) [reply src/main/java/com/zviproject/component/entity/MessageToDisplay.java]: Done

File: src/main/java/com/zviproject/component/entity/User.java (hunk 3 above)
Comment (ZVIproject) [reply src/main/java/com/zviproject/component/entity/User.java:24]: Done

File: src/main/java/com/zviproject/component/service/ChatDao.java (hunk 4 above)
Comment (ZVIproject) [reply src/main/java/com/zviproject/component/service/ChatDao.java]: Done

File: src/main/java/com/zviproject/component/service/ChatDao.java (hunk 5 above)
Comment (ZVIproject) [reply src/main/java/com/zviproject/component/service/ChatDao.java]: Done

**Review by OlegNovatskiy — APPROVED** [review]

**Review by ZVIproject — COMMENTED** [review]
done

**Pull request merged**